SEED-platform/seed
# 2134 blank col setting dev

------ This PR contains a migration -----

#### Any background context you want to provide?
See tagged issue

#### What's this PR do?
See commits

Note, when comparing this PR to #2135 which adds this feature to prod as a patch, this PR contains a few additional features:
- The column settings change preview includes this new setting
- This new setting is a sortable field in the column settings page
- The column setting page help text is includes info on this new setting
- The interaction between this new setting and merging geocoding results fields is that they are purposely handled separately and don't interact at all. Merging geocoding results happens before recognize_empty and doesn't consider recognize_empty at all. This is done to avoid pouring effort into what I'm considering to be a low impact edge case.

#### How should this be manually tested?
Toggle the Recognize Empty column setting. Import records that trigger merges and with Not Available or empty values in a particular field. When the setting is checked, see that empty values (truly blank, missing, or "Not Available") are treated like any other non-empty value during a merge.

**Notable caveat**: On import, if a known (by SEED) column is not provided/mapped to, SEED will think the incoming records have "empty" values for that column. This means, if recognize_blank is set to True for a particular column, and that column is not mapped to during import, any resulting merged records will have an empty value for that column/attribute.

#### What are the relevant tickets?
#2134 

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2020-03-11 at 2 15 14 PM" src="https://user-images.githubusercontent.com/30608004/76459965-c76eca80-63a2-11ea-9a73-ef260143ec58.png">
